### PR TITLE
refactor: use new API authentification

### PR DESCRIPTION
**Pull request by alexandre-abrioux**
Hi! First of all, thank you for creating this awesome tool!

This player works way better than the one provided by the (now deprecated) SoundCloud's Javascript SDK. I've been using your library in my project for a long time and it's been working great:
https://github.com/alexandre-abrioux/soundcloud-reverse  

Unfortunately, and because of the latest [SoundCloud's API security update](https://developers.soundcloud.com/blog/security-updates-api), I noticed this week that playback is not working anymore.

SoundCloud now requires an `Authorization` header containing an OAuth token on each request made to its API.

Furthermore it looks like legacy stream URLs like `https://api.soundcloud.com/tracks/[ID]/stream` are not playable anymore, because they also need this `Authorization` header, rendering them unusable (details bellow).

If you try to send an `Authorization` header to the stream with some audio tag hack (cf. this [Stackoverflow thread](https://stackoverflow.com/questions/9299189/send-custom-http-request-header-with-html5-audio-tag)), it won't work because those stream URLs do a 302 redirect to a CDN that won't accept the `Authorization` header, however the browser will send it, thus creating a CORS failure. There is no way to manually handle the redirection with xhr requests, even when using [fetch](https://developer.mozilla.org/en-US/docs/Web/API/fetch)'s `redirect=manual` option, as this is a [security feature from the spec](https://github.com/whatwg/fetch/issues/763).

This is why we have to use the new API endpoint [`/tracks/[ID]/streams`](https://developers.soundcloud.com/docs/api/explorer/open-api#/tracks/get_tracks__track_id__streams) (notice the `s` at the end of stream) to first get the CDN url and then use it as the audio source.

I'm using it now on my project and playback works again:

```bash
npm i --save https://github.com/alexandre-abrioux/soundcloud-audio.js.git#oauth
```

You just need to pass an OAuth token now instead of a `client_id`:

```js
var player = new SoundCloudAudio(token);
```

TL;DR: This is a refacto of the codebase that takes into consideration the new SoundCloud API update

### CODE REVIEW

**Comment by alexandre-abrioux** [comment]
I've pushed a [new commit](https://github.com/alexandre-abrioux/soundcloud-audio.js/commit/fb754ba89267de82a27a5e05920ed57623cdf207) to the `oauth` branch: we're now also using the OAuth token to fetch the stream, as fetching a stream without the Authorization header is no longer supported, and we're using the AAC HLS stream URL instead of the MP3 one (see [Soundcloud announcement](https://developers.soundcloud.com/blog/api-streaming-urls)).
